File: offer/findNumberIn2DArray.py
class Solution:
    def findNumberIn2DArray(self, matrix: List[List[int]], target: int) -> bool:
        if len(matrix) == 0 or len(matrix[0]) == 0: return False
        a = 0
        b = len(matrix[0]) - 1
        while b >= 0 and a < len(matrix):
            if matrix[a][b] == target: return True
            elif matrix[a][b] > target:
                b -= 1
            elif matrix[a][b] < target:
                a += 1
        return False
        
        

File: javaTest/javaCode/src/findNumberIn2DArray.java
class Solution {
    public boolean findNumberIn2DArray(int[][] matrix, int target) {
        int a = matrix.length - 1;
        int b = 0;
        while(a >= 0 && b < matrix[0].length && b >= 0){
            if(matrix[a][b] > target){
                a --;
            }else if(matrix[a][b] < target){
                b ++;
            }else if(matrix[a][b] == target){
                return true;
            }
        }
        return false;
    }
}